pyproject.toml

```diff
@@ -0,0 +1,52 @@
+[tool.ruff]
+
+# description of all rules are available on https://docs.astral.sh/ruff/rules/
+lint.select = ["D", "E", "F", "W", "C", "S", "I", "TCH", "SLOT", "RUF", "C90", "N", "YTT", "ASYNC", "TRIO"]
+
+# we need to check 'mood' of all docstrings, this needs to be enabled explicitly
+lint.extend-select = ["D401"]
+
+lint.ignore = []
+
+target-version = "py311"
+lint.pydocstyle.convention = "google"
+line-length = 100
+
+[tool.mypy]
+disable_error_code = ["union-attr", "return-value", "arg-type", "import-untyped"]
+ignore_missing_imports = true
+
+[tool.pdm]
+distribution = false
+
+[tool.pdm.dev-dependencies]
+dev = [
+    "black>=24.4.0",
+    "mypy>=1.8.0",
+    "ruff>=0.4.1",
+    "types-requests>=2.31.0"
+]
+
+[[tool.pdm.source]]
+type = "find_links"
+url = "https://download.pytorch.org/whl/cpu/torch_stable.html"
+name = "torch"
+
+[project]
+name = "lightspeed-rag-content"
+version = "0.1.0"
+description = "RAG content for OpenShift LightSpeed."
+authors = []
+dependencies = [
+    "PyYAML",
+    "huggingface_hub",
+    "llama_index",
+    "llama-index-vector-stores-faiss",
+    "llama-index-embeddings-huggingface",
+    "langchain_community",
+    "faiss-cpu",
+    "sentence_transformers",
```

Comment:
done

Reply:
@asamal4 pdm.lock already pins packages to exact versions, what's the benefit of doing this here as well?

Reply:
That's correct, because if x.y.2 is available, we should normally be upgrading to that and only say  `== x.y.1` if we have to stay on x.y.1. pyproject.toml is about constraints and pdm.lock is about pinning the exact versions.